libraries/standard/mqtt/include/mqtt_config.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file mqtt_config.h
+ * @brief This represents a template of the mqtt_config.h file for definining
+ * preprocessor macros that configure the the MQTT library.
+ */
+
+#ifndef MQTT_CONFIG_H_
+#define MQTT_CONFIG_H_
+
+/**
+ * @brief Definition of logging macros to map logging calls within MQTT
+ * library to application-specific logging implementation.
+ *
+ * Each logging macro represents a different logging level. Only define those
+ * logging macros whose corresponding logging level messages are desired in
+ * the MQTT library build.
+ *
+ * By default, logging is turned off, and no code is generated for logging calls
+ * in the MQTT library on compilation.
+ *
+ * @note The logging macros are called in MQTT library with paramerers wrapped in
+ * double parantheses to be ISO C89/C90 standard compliant. For an example
+ * implementation of the logging macros, refer to mqtt_config.h files of MQTT demo folder
+ * and the reference implementation of logging-stack in demos.
+ *
+ * #define LogError( X )           MyErrorLoggingFunction( X )
+ * #define LogWarn( X )            MyWarnLoggingFunction( X )
+ * #define LogInfo( X )            MyInfoLoggingFunction( X )
+ * #define LogDebug( X )           MyDebugLoggingFunction( X )
+ */
+
+/**
+ * @brief The maximum number of MQTT PUBLISH messages that may be pending
+ * acknowledgement at any time.
+ *
+ * QoS 1 and 2 MQTT PUBLISHes require acknowledgement from the server before
+ * they can be completed. While they are awaiting the acknowledgement, the
+ * client must maintain information about their state. The value of this
+ * macro sets the limit on how many simultaneous PUBLISH states an MQTT
+ * context maintains.
```

Comment:
I think we may be missing the information here that we maintain the incoming and outgoing publishes separately. Total number of Publishes including both incoming and outgoing publishes are 2* the config value.

Reply:
Good point, have updated the comment